nixos/modules/services/mail/stalwart-mail.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.stalwart-mail;
+  configFormat = pkgs.formats.toml { };
+  configFile = configFormat.generate "stalwart-mail.toml" cfg.settings;
+  dataDir = "/var/lib/stalwart-mail";
+
+in {
+  options.services.stalwart-mail = {
+    enable = mkEnableOption (mdDoc "the Stalwart all-in-one email server");
+    package = mkPackageOptionMD pkgs "stalwart-mail" { };
+
+    settings = mkOption {
+      inherit (configFormat) type;
+      default = { };
+      description = mdDoc ''
+        Configuration options for the Stalwart email server.
+        See <https://stalw.art/docs/> for available options.
+
+        By default, the module is configured to store everything locally.
+      '';
```

Comment:
I was thinking about adding an example attribute that doesn't involve "in-memory" directory for the users, something with sqlite. But I'm a bit limited in time at the moment.
The main idea was to try to capture things like "domain is needed to actually run this".

Reply:
The default uses sqlite.
Only the VM test uses the static configuration ("in-memory").
For more specific examples, I think the wiki would be a better place.
The main documentation of Stalwart cannot be ignored either.

Reply:
Sounds good to leave this in the wiki then.